ops/charm.py

```diff
@@ -1105,12 +1105,13 @@ class CollectStatusEvent(LifecycleEvent):
 
     The order of priorities is as follows, from highest to lowest:
 
-    * error
     * blocked
     * maintenance
     * waiting
     * active
-    * unknown
+
+    It is an error to call :meth:`add_status` with an instance of
+    :class:`ErrorStatus` or :class:`UnknownStatus`.
```

Comment:
I would use the `Raises:` style to document this (which will also mean we can get rid of that line with all the ignores, I think, as a side-effect bonus).

Something like:

```
Raises:
    InvalidStatusError: if :class:`ErrorStatus` or :class:`UnknownStatus` are added
```

Reply:
I'm drawing a blank, sorry, which is the line with the ignores?

Reply:
It's [right above](https://github.com/canonical/operator/blob/98a4e4f44d37d3361a253bc1eff49eee2ca6d140/ops/charm.py#L1145) where you would put the `Raises:` lines.

Reply:
I wonder if the `raises` would be best in the `add_status` docstring, but then that isn't so good for addressing the "why aren't all six statuses listed in the priority order" question ...
![image](https://github.com/user-attachments/assets/afb82657-3359-43ae-aeab-1d0c57610dcd)
vs
![image](https://github.com/user-attachments/assets/0c07b7e9-a764-4717-ad4b-591a19572ef0)

Reply:
I think the "raises" belongs on the `add_status` docstring, as that's what will actually raise. I think it's fine without those in the bulleted list -- in fact, perhaps clearer.

Reply:
Oh, good point that these are two different docstrings, I somehow missed that.

Should `Raises: ...` should go into `add_status`? We try to be complete in documenting the exceptions that can be raised, but only for those that we expect the charm may want to catch (typically: something unexpected has gone wrong). We don't document the ones like `TypeError` that are really a bug in the charm. I think this does fall into that category (unless the charm was getting the status from somewhere external but that seems very unlikely and should be handed in the charm code).

So I take back my "do this with Raises:" suggestion, sorry, and I would stick with what you originally had.

Reply:
Funny timing, I just pushed the change removing the sentences from the class documentation and adding `Raises:` to `add_status`. I'll just revert that to what I had before and merge.

I had a look at removing those `noqa` directives, but it seems like `ruff` really doesn't like `return` in a code sample in a docstring.